nexus/src/app/rack.rs

```diff
@@ -206,20 +206,38 @@ impl super::Nexus {
         );
 
         let silo_name = &request.recovery_silo.silo_name;
-        let dns_records = request
+        // Records that should be present at the rack-internal zone apex -
```

Comment:
It's important to know that this code path only affects the contents of the first internal DNS zone for newly-initialized racks.  None of this will run for deployed racks.  I assume that's fine?

(We probably should have an end-to-end test that the internal DNS contents doesn't change after the initial blueprint is executed.  We have a similar end-to-end test that generating a new blueprint from the initial one makes no changes.)

Relatedly: should we be using `blueprint_internal_dns_config()` like you're doing with the external zone below?

Reply:
> None of this will run for deployed racks. I assume that's fine?

yeah, the changes in `rack.rs` are really just in service of minimizing the delta between state just after RSS and what the initial blueprint describes. except that as you noted below, the records don't actually get added to the internal update, which is a bug.

> Relatedly: should we be using `blueprint_internal_dns_config()` like you're doing with the external zone below?

probably! i didn't make that change here because the initial internal DNS records are created by sled-agent, and i'd want to either compare those more closely. or, have an end-to-end test like you mention that checks that the records as created by sled-agent are the same as the initial blueprint.